mlflow/models/evaluation/base.py

```diff
@@ -1991,36 +2017,9 @@ def pred_sample(eval_df, _builtin_metrics, _artifacts_dir):
                 "the desired configuration there.",
                 error_code=INVALID_PARAMETER_VALUE,
             )
-    elif model is None:
-        # Evaluating a static dataset
-        if isinstance(data, pd.DataFrame):
-            # If data is a pandas dataframe, predictions must be specified
-            if predictions is None:
-                raise MlflowException(
-                    message="The model output must be specified in the predictions "
-                    "parameter when model=None.",
-                    error_code=INVALID_PARAMETER_VALUE,
-                )
-        elif isinstance(data, mlflow.data.pandas_dataset.PandasDataset):
-            # If data is a mlflow PandasDataset, data.predictions must be specified
-            if data.predictions is None:
-                raise MlflowException(
-                    message="The predictions parameter must be specified with the provided "
-                    "PandasDataset when model=None. For example: "
-                    "`data = mlflow.data.from_pandas(df=X.assign(y=y), predictions='y')`",
-                    error_code=INVALID_PARAMETER_VALUE,
-                )
-        else:
-            # Other data formats are not supported
-            raise MlflowException(
-                message="The data must be a pandas dataframe or mlflow.data.pandas_dataset."
-                "PandasDataset when model=None.",
-                error_code=INVALID_PARAMETER_VALUE,
-            )
```

Comment:
Move this into the default evaluator. This will allow other evaluators to work without requiring `predictions` to be specified